### PR TITLE
Bugfix/lr finder

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/1648 where the learning rate finder could not be used by users that prefer to pass `val_dataloaders` directly to the different methods.

Fixed https://github.com/PyTorchLightning/pytorch-lightning/issues/1650 where the learning rate finder could not be used, when a `EarlyStopping` callback is passed to trainer. 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/lr_finder.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def _run_lr_finder_internally(self, model: LightningModule):
     def lr_find(self,
                 model: LightningModule,
                 train_dataloader: Optional[DataLoader] = None,
+                val_dataloaders: Optional[DataLoader] = None,
```
Comment (Borda) [inline pytorch_lightning/trainer/lr_finder.py:56]: do we need to add also `test_dataloaders`

Comment (SkafteNicki) [reply pytorch_lightning/trainer/lr_finder.py:56]: no, the bug happens due to this check:
https://github.com/PyTorchLightning/pytorch-lightning/blob/142bc0230e228cd2e851481e5a07069e7d198655/pytorch_lightning/trainer/trainer.py#L1009-L1012
which is only present for the validation dataloader.

**Comment by SkafteNicki** [comment]
The idea behind dumping and restoring trainer args is to take advantage of `fit` during the learning rate search (alternative would be to write a new training loop). I agree that this is a fragile feature, however for me the clear advantages of calling `fit` internally outweigh these disadvantages

**Comment by awaelchli** [comment]
@SkafteNicki it makes sense to use fit, that's very good :) we just have to keep it in mind when we add new trainer args :) 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed Horovod distributed backend to set the `root_gpu` property ([#1669](https://github.com/PyTorchLightning/pytorch-lightning/pull/1669))
 
+- Fixed bugs that prevent lr finder to be used together with early stopping and validation dataloaders ([1676](https://github.com/PyTorchLightning/pytorch-lightning/pull/1676))
```
Comment (Borda) [inline CHANGELOG.md:27]:
```suggestion
- Fixed bugs that prevent lr finder to be used together with early stopping and validation dataloaders ([#1676](https://github.com/PyTorchLightning/pytorch-lightning/pull/1676))
```
